test/range_test.exs

```diff
@@ -557,5 +655,107 @@ defmodule SourcerorTest.RangeTest do
                end: [line: 4, column: 7]
              }
     end
+
+    test "captures" do
+      assert to_range(~S"&foo/1") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 7]
+             }
+
+      assert to_range(~S"&Foo.bar/1") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 11]
+             }
+
+      assert to_range(~S"&__MODULE__.Foo.bar/1") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 22]
+             }
+    end
+
+    test "captures with arguments" do
+      assert to_range(~S"&foo(&1, :bar)") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 15]
+             }
+
+      assert to_range(~S"& &1.foo") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 9]
+             }
+
+      assert to_range(~S"& &1") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 5]
+             }
+
+      # This range currently ends on column 5, though it should be column 6,
+      # and appears to be a limitation of the parser, which does not include
+      # any metadata about the parens. That is, this currently holds:
+      #
+      #     Sourceror.parse_string!("& &1") == Sourceror.parse_string!("&(&1)")
+      #
+      # assert to_range(~S"&(&1)") == %{
+      #          start: [line: 1, column: 1],
+      #          end: [line: 1, column: 6]
+      #        }
+    end
+
+    test "arguments in captures" do
+      {:&, _, [{:&, _, _} = arg]} = Sourceror.parse_string!(~S"& &1")
+
+      assert Sourceror.Range.get_range(arg) == %{
+               start: [line: 1, column: 3],
+               end: [line: 1, column: 5]
+             }
+    end
+
+    test "Access syntax" do
+      assert to_range(~S"foo[bar]") == %{
+               start: [line: 1, column: 1],
+               end: [line: 1, column: 9]
+             }
+
+      {{:., _, [Access, :get]} = access, _, _} = Sourceror.parse_string!(~S"foo[bar]")
+
+      assert Sourceror.Range.get_range(access) == %{
+               start: [line: 1, column: 4],
+               end: [line: 1, column: 9]
+             }
+    end
+
+    test "should not raise on any three-element tuple parsed by parse_string" do
+      for relative_path <- Path.wildcard("lib/*/**.ex") do
+        assert_can_get_ranges(relative_path)
+      end
+    end
```

Comment:
This is a cool trick, and while for now it works it might be annoying for future PRs that unintendedly add code that braks range calculation.

I think your idea of bringing the examples from the elixir tree-sitter corpus sounds good, so we can add that in another PR 👍

Reply:
> I think your idea of bringing the examples from the elixir tree-sitter corpus sounds good, so we can add that in another PR 👍

Sounds good! So just to be clear: leave this for now, and then we can strip it out when we pull in the better corpus of syntax examples?

Reply:
Yup!